docs/api/http.md

```diff
@@ -4,6 +4,95 @@ outline: deep
+
+Then you switch into a live mode to use long-polling to recieve real-time updates. We'll go over these steps in more detail below. First a note on the data that the endpoint returns.
+
+### Shape Log
+
+When you sync a shape from Electric, you get the data in the form of a log of logical database operations. This is the **Shape Log**.
+
+The `offset` that you see in the messages and provide as the `?offset=...` query parameter in your request identifies a position in the log. The messages you see in the response are shape log entries (the ones with `value`s and `action` headers) and control messages (the ones with `control` headers).
+
+The Shape Log is similar conceptually to the logical replication stream from Postgres. Except that instead of getting all the database operations, you're getting the ones that affect the data in your Shape. It's then the responsibility of the client to consume the log and materialize out the current value of the shape.
+
+<figure>
+  <a href="/img/api/shape-log.jpg">
+    <img srcset="/img/api/shape-log.sm.png 1064w, /img/api/shape-log.png 1396w"
+        sizes="(max-width: 767px) 600px, 1396px"
+        src="/img/api/shape-log.png"
+        alt="Shape log flow diagramme"
+    />
+  </a>
+  <figcaption className="figure-caption text-end">
+    Shape log flow diagramme.
+  </figcaption>
+</figure>
+
+### Initial sync request
+
+When you make an initial sync request, with `offset=-1`, you're telling the server that you want the whole log, from the start for a given shape.
+
+When a shape is first requested, Electric queries Postgres for the data and populates the log by turning the query results into insert operations. This allows you to sync shapes without having to pre-define them. Electric then streams out the log data in the response.
+
+Sometimes a log can fit in a single response. Sometimes it's too big and requires multiple requests. In this case, the first request will return a batch of data and an `x-electric-chunk-last-offset` header. An HTTP client should then continue to make requests setting the `offset` parameter to the this header value. This allows the client to paginate through the shape log until it has recieved all the current data.
+
+### Control messages
+
+The client will then recieve an `up-to-date` control message at the end of the response data:
+
+```json
+{"headers": {"control": "up-to-date"}}
+```
+
+This indicates that the client has all the data that the server was aware of when fulfilling the request. The client can then switch into live mode to recieve real-time updates.
+
+::: info Must-refetch
+Note that the other control message is `must-refetch` which indicates that the client must throwaway their local shape data and re-sync from scratch:
+
+```json
+{"headers": {"control": "must-refetch"}}
+```
+:::
+
+### Live mode
+
+Once a client is up-to-date, it can switch to live mode to recieve real-time updates, by making requests with `live=true`, an `offset` and a `shape_id`, e.g.:
```

Comment:
```suggestion
Once a client is up-to-date, it can switch to live mode to receive real-time updates, by making requests with `live=true`, an `offset` and a `shape_id`, e.g.:
```